### ISSUE TITLE
Wanted to confirm my understanding as SC 1.4.11 applies to hover

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#896]
Based on reading the SC and the associated github comments -- only the parts needed for identification have to meet contrast requirements for the hovered state.

* A unchecked checkbox when hovered has a border < 3:1 to adjacent colors - FAIL
* A checked checkbox when hovered has a border < 3:1 but the checkmark > 3:1 - PASS
* A button with text when hovered has a border < 3:1 to adjacent colors - PASS - borders are not required to identify a button according to this working group

There continues to be misunderstanding in the community around the hovered and focused states and this SC.  For example, this SC does not require a focus ring to meet 3:1, etc.

**Comment by patrickhlauke** [comment]
i would say the "hover" aspect here is a bit of a red herring, and that generally the point is:

- the SC doesn't require that the contrast between any of the colors chosen for individual states need to be 3:1 amongs each other (i.e. it doesn't say something like "when a component is hovered, or focused, or selected, its color must contrast 3:1 against its normal default look" or something). only exception: if there are controls that butt up against each other, with no gap, making the focused or hovered or selected color *adjacent* to another component's default color - but technically, even a single pixel gap makes them non-adjacent.

- what the SC does say is that if the non-text element (the outline, or border, or extra icon, or whatever) is essential to identify a component itself, or its state, then that must have a 3:1 contrast - and this applies to all possible states of a component.

> For example, this SC does not require a focus ring to meet 3:1

the understanding doc does explicitly call this out though "In combination with 2.4.7 Focus Visible, the visual focus indicator for a component must have sufficient contrast against the adjacent background when the component is focused". and, from the normative language of the SC, having a focus ring IS required to "identify [...] state".

what this SC doesn't cover, in my view anyway, is a scenario where focus indication is simply a change of color and luminosity - say all regular controls are blue, but the focused control uses a different color (but otherwise no other visual indication) - as long as focused and non-focused controls aren't adjacent, there's no prescription that says anything about how contrasty the focused/unfocused colors need to be between them, only that taken individually (if they're required to identify that something IS a control) they're contrasty enough against the background of the page. (though it may be argued that this would also violate "color alone", but i do recall some fairly hairsplitting discussion about changes in both hue/saturation/lightness counting or not counting as a "color alone" case, I forget which one was argued in the end).

**Comment by JAWS-test** [comment]
I think the big difference between hover and focus is that there is an own SC for focus and not for hover. I.e. the WCAG does not require an element to change at all for hover. However, it requires a visual change while maintaining focus. 
* If the colors change in hover so that the element is no longer sufficiently visible (e.g. contrast between foreground and background < 3:1 or 4,5:1), it is a violation of 1.4.11 or 1.4.3.
* If the hover effect itself is not sufficiently visible (e.g. the background color changes imperceptibly), this does not violate any SC, because hover effects are not required. Unless you say that's a color coding (1.4.1). 
* But if the hover effect is indicated by an icon and the icon does not have enough contrasts, it would be a violation of 1.4.11.

I find the last two points somewhat inconsistent. Either I misunderstand or there is still a need for improvement.

I also find it a pity that SC 2.4.7 does not make any statements about the minimum contrast required for the focus. I.e. if the background colour changes only imperceptibly with focus, the question is again whether this would be a violation of 2.4.7 or 1.4.1. or no violation.

**Comment by alastc** [comment]
There's lots of scenarios/cases to this, I've just been reminding myself of the previous conversations and changes:

- PR #550 with lots of conversation in it.
- Survey: https://www.w3.org/2002/09/wbs/35422/technique-approvals3/results#xq22
- Minutes: https://www.w3.org/2018/12/18-ag-minutes.html#item03

> we need to clarify what is and is not covered in regard to hover.

My running assumption has been that the control should maintain contrast during hover (and other) state, i.e. not 'disappear' by going to a lighter color. If the control doesn't have contrast by default then that's also a fail.

I made an overly blanket statement on the call, as Pat outlined above there are some scenarios where 1.4.11 comes into play, but I don't think that's enough. There is a weakness in using 'adjacent' colors for a dynamic change, which is why I think the focus-visible-enhanced SC is useful.

> We really need to get in clear agreement with a table listing scenarios as this is causing mass confusion in the community and it seems even among people

Agree, I've been a bit swamped, can anyone help with this?

**Comment by JAWS-test** [comment]
> We really need to get in clear agreement with a table listing scenarios 

Before such a table is written, however, agreement would have to be reached on what 1.4.11 covers and how 1.4.11 interacts with other criteria. I am thinking, for example, of questions such as 
* should the same problem belong to different criteria depending on whether there is a border between 2 colors or not (https://github.com/w3c/wcag/issues/875)?
* which contrast at 1.4.1 is sufficient to distinguish colors (https://github.com/w3c/wcag/issues/873)?
* what visual changes must be made to ensure that the 3:1 difference in contrast is sufficient and to which elements this requirement applies to comply with 1.4.1 (https://github.com/w3c/wcag/issues/871)?
* it is considered problematic that a poorly visible hover state is only a problem at 1.4.11 if it is displayed with an additional visual element, but not if only the colors change (then it's not a problem at all, https://github.com/w3c/wcag/issues/899)?
* it is considered problematic that a poorly visible focus state is only a problem at 1.4.11 if it is displayed with an additional visual element (e.g. focus ring), but not if only the colors change (then it is a problem at SC 2.4.7, but without any contrast requirements)

**Comment by alastc** [comment]
In terms of the original question in the thread, I think that has been answered in the updated to the understanding doc in #931, so closing this issue.
